### PR TITLE
Rerank Process

### LINKED ISSUE #14: Rerank process
# Description
Wrap [`rerank.netcdf.wrapper`](https://github.com/pacificclimate/ClimDown/blob/master/R/rerank.R#L58) into `pywps` process.

# Requested Action
Use `rpy2` to translate between `R` and `python`.

# Alternatives
None.

### PR DESCRIPTION
**Pull request by cairosanders**
### Rerank
closes #14 

Last step of the bccaq process is [rerank](https://github.com/pacificclimate/ClimDown/blob/master/R/rerank.R). It takes some of the usual input as well as an R object `analogues.rds` (output from constructed analogues (CA) step) and the output from the QDM step, `qdm_file` 

You can test with `make test-notebooks-custom` on port `30734` and `pytest tests`

### CODE REVIEW

File: docs/source/notebooks/wps_rerank_demo.ipynb [hunk 1]
```diff
@@ -0,0 +1,203 @@
+{
```
Comment (nikola-rados) [inline docs/source/notebooks/wps_rerank_demo.ipynb:1]: <p>This extra url is causing issues in <code>make</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pacificclimate/chickadee/pull/27/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

File: chickadee/processes/wps_rerank.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+            logger,
+            log_level=loglevel,
+            process_step="start",
+        )
+
+        (
+            obs_file,
+            varname,
+            out_file,
+            num_cores,
```
Comment (nikola-rados) [inline chickadee/processes/wps_rerank.py:88]: Does `num_cores` have a default value? I notice you are providing cores in the test file.

**Review by nikola-rados — APPROVED** [review]
This looks good to me. `pytest` passes but there is a small mistake to fix in the notebook.

**Review by sum1lim — APPROVED** [review]
The PR looks good after fixing the notebooks as @nikola-rados suggested. Nice work!

File: docs/source/notebooks/wps_rerank_demo.ipynb (hunk 1 above)
Comment (cairosanders) [reply docs/source/notebooks/wps_rerank_demo.ipynb:1]: Oh woops!

File: chickadee/processes/wps_rerank.py (hunk 2 above)
Comment (cairosanders) [reply chickadee/processes/wps_rerank.py:88]: It does, I just thought it would be good to test :woman_shrugging:

Comment (cairosanders) [reply chickadee/processes/wps_rerank.py:88]: Oh You're talking pytest, I was thinking demo. I guess it's not necessary for the pytest, you're right

**Pull request merged**